huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_consumer_group.go

```diff
@@ -0,0 +1,300 @@
+		"broadcast":      utils.ValueIngoreEmpty(d.Get("broadcast")),
+		"brokers":        utils.ValueIngoreEmpty(d.Get("brokers")),
+		"name":           utils.ValueIngoreEmpty(d.Get("name")),
+		"retry_max_time": utils.ValueIngoreEmpty(d.Get("retry_max_time")),
+	}
+	return bodyParams
+}
+
+func resourceDmsRocketMQConsumerGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	updateRocketmqConsumerGrouphasChanges := []string{
+		"enabled",
+		"broadcast",
+		"retry_max_time",
+	}
+
+	if d.HasChanges(updateRocketmqConsumerGrouphasChanges...) {
+		// updateRocketmqConsumerGroup: update DMS rocketmq consumer group
+		var (
+			updateRocketmqConsumerGroupHttpUrl = "v2/{project_id}/instances/{instance_id}/groups/{group}"
+			updateRocketmqConsumerGroupProduct = "dms"
+		)
+		updateRocketmqConsumerGroupClient, err := config.NewServiceClient(updateRocketmqConsumerGroupProduct, region)
+		if err != nil {
+			return diag.Errorf("error creating DmsRocketMQConsumerGroup Client: %s", err)
+		}
+
+		parts := strings.SplitN(d.Id(), "/", 2)
+		if len(parts) != 2 {
+			return diag.Errorf("invalid id format, must be <instance_id>/<user>")
+		}
+		instanceID := parts[0]
+		name := parts[1]
+		updateRocketmqConsumerGroupPath := updateRocketmqConsumerGroupClient.Endpoint + updateRocketmqConsumerGroupHttpUrl
+		updateRocketmqConsumerGroupPath = strings.ReplaceAll(updateRocketmqConsumerGroupPath, "{project_id}", updateRocketmqConsumerGroupClient.ProjectID)
+		updateRocketmqConsumerGroupPath = strings.ReplaceAll(updateRocketmqConsumerGroupPath, "{instance_id}", instanceID)
+		updateRocketmqConsumerGroupPath = strings.ReplaceAll(updateRocketmqConsumerGroupPath, "{group}", name)
+
+		updateRocketmqConsumerGroupOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			OkCodes: []int{
+				204,
+			},
+		}
+		updateRocketmqConsumerGroupOpt.JSONBody = utils.RemoveNil(buildUpdateRocketmqConsumerGroupBodyParams(d, config))
+		_, err = updateRocketmqConsumerGroupClient.Request("PUT", updateRocketmqConsumerGroupPath, &updateRocketmqConsumerGroupOpt)
+		if err != nil {
+			return diag.Errorf("error updating DmsRocketMQConsumerGroup: %s", err)
+		}
+	}
+
+	return resourceDmsRocketMQConsumerGroupRead(ctx, d, meta)
+}
+
+func buildUpdateRocketmqConsumerGroupBodyParams(d *schema.ResourceData, config *config.Config) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"broadcast":      utils.ValueIngoreEmpty(d.Get("broadcast")),
+		"retry_max_time": utils.ValueIngoreEmpty(d.Get("retry_max_time")),
+	}
+	enabled := utils.ValueIngoreEmpty(d.Get("enabled"))
+	if enabled != nil {
+		bodyParams["enabled"] = enabled
+	}
+	return bodyParams
+}
+
+func resourceDmsRocketMQConsumerGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+
+	var mErr *multierror.Error
```

Comment:
remove this line to line:248

Reply:
the code is automatically generated, the style is unified